plugins/Calendars/src/JapaneseCalendar.cpp

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2022 Georg Zotti
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA  02110-1335, USA.
+ */
+
+#include "StelTranslator.hpp"
+#include "JapaneseCalendar.hpp"
+//#include "GregorianCalendar.hpp"
```

Comment:
Why is this commented out? I get compilation failures unless I uncomment this line and similar ones in two other files (and I wonder why the CI builds don't fail).

Reply:
interesting. The Clangd syntax checker in QtCreator recommended to remove the include as it is not used.
What is the actual error?

Reply:
```
/home/ruslan/src/stellarium/plugins/Calendars/src/JapaneseCalendar.cpp:201:17: error: ‘GregorianCalendar’ has not been declared
  201 |  const int year=GregorianCalendar::gregorianYearFromFixed(qRound(floor(rd_t)));
      |                 ^~~~~~~~~~~~~~~~~
make[2]: *** [plugins/Calendars/src/CMakeFiles/Calendars-static.dir/build.make:583: plugins/Calendars/src/CMakeFiles/Calendars-static.dir/JapaneseCalendar.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/ruslan/src/stellarium/plugins/Calendars/src/VietnameseCalendar.cpp: In static member function ‘static StelLocation VietnameseCalendar::vietnameseLocation(double)’:
/home/ruslan/src/stellarium/plugins/Calendars/src/VietnameseCalendar.cpp:243:17: error: ‘GregorianCalendar’ has not been declared
  243 |  const int year=GregorianCalendar::gregorianYearFromFixed(qRound(floor(rd_t)));
      |                 ^~~~~~~~~~~~~~~~~
make[2]: *** [plugins/Calendars/src/CMakeFiles/Calendars-static.dir/build.make:609: plugins/Calendars/src/CMakeFiles/Calendars-static.dir/VietnameseCalendar.cpp.o] Error 1
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp: In static member function ‘static StelLocation KoreanCalendar::koreanLocation(double)’:
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:259:11: error: ‘GregorianCalendar’ has not been declared
  259 |  if (rd_t<GregorianCalendar::fixedFromGregorian({1908, GregorianCalendar::april, 1}))
      |           ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:259:56: error: ‘GregorianCalendar’ has not been declared
  259 |  if (rd_t<GregorianCalendar::fixedFromGregorian({1908, GregorianCalendar::april, 1}))
      |                                                        ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:261:17: error: ‘GregorianCalendar’ has not been declared
  261 |  else if ((rd_t<GregorianCalendar::fixedFromGregorian({1912, GregorianCalendar::january, 1})) ||
      |                 ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:261:62: error: ‘GregorianCalendar’ has not been declared
  261 |  else if ((rd_t<GregorianCalendar::fixedFromGregorian({1912, GregorianCalendar::january, 1})) ||
      |                                                              ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:262:12: error: ‘GregorianCalendar’ has not been declared
  262 |    ((rd_t>=GregorianCalendar::fixedFromGregorian({1954, GregorianCalendar::march, 21})) &&
      |            ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:262:57: error: ‘GregorianCalendar’ has not been declared
  262 |    ((rd_t>=GregorianCalendar::fixedFromGregorian({1954, GregorianCalendar::march, 21})) &&
      |                                                         ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:263:11: error: ‘GregorianCalendar’ has not been declared
  263 |     (rd_t<GregorianCalendar::fixedFromGregorian({1961, GregorianCalendar::august, 10}))))
      |           ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:263:56: error: ‘GregorianCalendar’ has not been declared
  263 |     (rd_t<GregorianCalendar::fixedFromGregorian({1961, GregorianCalendar::august, 10}))))
      |                                                        ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp: In static member function ‘static int KoreanCalendar::koreanNewYearInGregorianYear(int)’:
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:446:17: error: ‘GregorianCalendar’ has not been declared
  446 |  return gYear+1-GregorianCalendar::gregorianYearFromFixed(ChineseCalendar::chineseEpoch)-304;
      |                 ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp: In static member function ‘static int KoreanCalendar::koreanAge(QVector<int>, int)’:
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:470:20: error: ‘GregorianCalendar’ has not been declared
  470 |  const int gYearRD=GregorianCalendar::gregorianFromFixed(rd).at(0);
      |                    ^~~~~~~~~~~~~~~~~
/home/ruslan/src/stellarium/plugins/Calendars/src/KoreanCalendar.cpp:472:23: error: ‘GregorianCalendar’ has not been declared
  472 |  const int gYearBirth=GregorianCalendar::gregorianFromFixed(rdBirth).at(0);
      |                       ^~~~~~~~~~~~~~~~~
```

Reply:
Ah, I use the Precompiled Headers feature and have Gregorian included in the PCH list. 
You are right, these headers are required in this case. Fixed in master.